test/e2e/external_gateways.go

```diff
@@ -798,13 +887,22 @@ var _ = ginkgo.Context("BFD", func() {
 // setupGatewayContainers sets up external containers, adds routes to the nodes, sets up udp / tcp listeners
 // that return the container's hostname.
 // All its needed for namespace / pod gateway tests.
-func setupGatewayContainers(f *framework.Framework, nodes *v1.NodeList, gwContainer1, gwContainer2, srcPodName string, updPort, tcpPort, numOfIPs int, postCreations ...func(string)) (gatewayTestIPs, gatewayTestIPs) {
+func setupGatewayContainers(f *framework.Framework, nodes *v1.NodeList, gwContainers []string, srcPodName string, updPort, tcpPort, numOfIPs int, postCreations ...func(string)) (gatewayTestIPs, gatewayTestIPs) {
 	addressesv4 := gatewayTestIPs{targetIPs: make([]string, 0)}
 	addressesv6 := gatewayTestIPs{targetIPs: make([]string, 0)}
-
 	ginkgo.By("Creating the gateway containers for the icmp test")
-	addressesv4.gatewayIPs[0], addressesv6.gatewayIPs[0] = createClusterExternalContainer(gwContainer1, externalContainerImage, []string{"-itd", "--privileged", "--network", externalContainerNetwork}, []string{})
-	addressesv4.gatewayIPs[1], addressesv6.gatewayIPs[1] = createClusterExternalContainer(gwContainer2, externalContainerImage, []string{"-itd", "--privileged", "--network", externalContainerNetwork}, []string{})
+	if externalContainerNetwork == "host" {
+		_, _ = createClusterExternalContainer(gwContainers[0], externalContainerImage, []string{"-itd", "--privileged", "--network", externalContainerNetwork}, []string{})
+		addressesv4.gatewayIPs = append(addressesv4.gatewayIPs, externalContainerIPv4)
```

Comment:
`createClusterExternalContainer` returns the addresses by docker inspecting the container's `.NetworksSettings.Networks`, which has an empty address when using `host` network for the container.
since a host can have multiple ips, we require a specific ip to be set from outside so we know how to reach that container.